src/lib/gundb/UserStorageClass.js

```diff
@@ -2530,10 +2530,19 @@ export class UserStorage {
     const { profile, _getProfileFields } = this
     let profileFields = await profile.then(_getProfileFields)
 
-    logger.debug('Deleting profile fields', profileFields)
+    const filterFields = field => !['avatar', 'smallAvatar'].includes(field)
 
+    logger.debug('Deleting profile fields', profileFields)
```

Comment:
there's a code repetition (in the part related to the error handling)
declare the following function:

```
import { first } from 'lodash'

const deleteField = async field => {
  if (!field.includes('avatar')) {
    // if not an 'avatar', 'smallAvatar' - set to '*****'
    await this.setProfileFieldPrivacy(field, 'private')
    return
  }

  if (field === 'avatar') {
    // as removeAvatar nullifies both field, we'll call it just once,
    // when we met 'avatar' field in the list. for 'smallAvatar' we won't do anything
    await this.removeAvatar()  
  }
}
```

and use it inside `retry`:

```javascript
retry(() => deleteField(field), 1)
```

Also you could adjust existing error handler and set corresponding error message if exception happened during 'avatar' field removal:

```javascript
message = 'Some error occurred during' + (field === 'avatar' ? 'deleting avatar' : 'setting the privacy to the field')
```

Also it would be good to specify which field failed exactly:

```javascript
error = new Error(`Deleting profile field '${field}' failed`)
```